### PR TITLE
document Helm self-signer, cert rotation

### PR DESCRIPTION
**Pull request by taroface**
Fixes https://github.com/cockroachdb/helm-charts/issues/125

Added docs for:

- Helm self-signer
- Helm cert rotation
- Migration to Helm self-signer

Also updated the Helm support callout in `deploy-cockroachdb-kubernetes.html`.

Note: This updates the v21.1 docs. A separate, broader refactor of the K8s docs area for v21.2 is in progress and will include these updates.

### CODE REVIEW

**Review by jseldess — APPROVED** [review]
LGTM, with a few suggestions.

**Comment by jseldess** [comment]
There are also some broken links that need to be fixed before netlify will build a test site: 

```
5:07:05 PM: running in concurrent mode, this is experimental
5:07:09 PM: docs/v21.1/deploy-cockroachdb-with-kubernetes.html
5:07:09 PM:   target does not exist --- docs/v21.1/deploy-cockroachdb-with-kubernetes.html --> secure-cockroachdb-kubernetes.html?filters=helm
5:07:10 PM: docs/v21.1/orchestrate-a-local-cluster-with-kubernetes.html
5:07:10 PM:   target does not exist --- docs/v21.1/orchestrate-a-local-cluster-with-kubernetes.html --> secure-cockroachdb-kubernetes.html?filters=helm
```

File: _includes/v21.1/orchestration/start-cockroachdb-helm-secure.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 {{site.data.alerts.callout_danger}}
-The CockroachDB Helm chart is undergoing maintenance for compatibility with Kubernetes versions 1.17 through 1.21 (the latest version as of this writing). No new feature development is currently planned. For new production and local deployments, we currently recommend using a manual configuration (**Configs** option). If you are experiencing issues with a Helm deployment on production, contact our [Support team](https://support.cockroachlabs.com/).
+The CockroachDB Helm chart is compatible with Kubernetes versions 1.17 through 1.22 (the latest version as of this writing). No new feature development is currently planned. If you are experiencing issues with a Helm deployment on production, contact our [Support team](https://support.cockroachlabs.com/).
```
Comment (jseldess) [inline _includes/v21.1/orchestration/start-cockroachdb-helm-secure.md:2]: Not sure why we need to mention 1.17. I think we could just say "compatible with Kubernetes 1.22 and earlier."

Also, I'd add, "However, no new feature development...". That however seems like the main point of the warning.

File: _includes/v21.1/orchestration/start-cockroachdb-helm-secure.md [hunk 2]
```diff
@@ -1,5 +1,9 @@
...
+{{site.data.alerts.end}}
+
+{{site.data.alerts.callout_danger}}
+If you are already running a secure Helm deployment on Kubernetes 1.22 and later, you must migrate away from using the Kubernetes CA for cluster authentication. For details, see [Operate CockroachDB on Kubernetes](operate-cockroachdb-kubernetes.html?filters=helm#migration-to-self-signer).
```
Comment (jseldess) [inline _includes/v21.1/orchestration/start-cockroachdb-helm-secure.md:6]: Not sure why this is a separate warning. Feels like it should be combined with the one above.

File: _includes/v21.1/orchestration/start-cockroachdb-helm-secure.md (hunk 1 above)
Comment (jseldess) [reply _includes/v21.1/orchestration/start-cockroachdb-helm-secure.md:2]: Seems like we should take another look at the Limitations section as well and make sure we're saying the same things there: https://www.cockroachlabs.com/docs/stable/deploy-cockroachdb-with-kubernetes.html#limitations

**Review by taroface — COMMENTED** [review]
TFTR!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/11206#-:-MicdtSODej0U--4oRrK:b-814g8k)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @jseldess and @udnay)

___
*[_includes/v21.1/orchestration/start-cockroachdb-helm-secure.md, line 2 at r1](https://reviewable.io/reviews/cockroachdb/docs/11206#-MiYAKrkoeQo28MZLu-g-r1-2:-MicdcTP4dvk-Ztwk08S:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/6088ab2fafe8a65fbed4ffd187e259216cbd1f50/_includes/v21.1/orchestration/start-cockroachdb-helm-secure.md#L2)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Seems like we should take another look at the Limitations section as well and make sure we're saying the same things there: https://www.cockroachlabs.com/docs/stable/deploy-cockroachdb-with-kubernetes.html#limitations
</blockquote></details>

Done.
___
*[_includes/v21.1/orchestration/start-cockroachdb-helm-secure.md, line 6 at r1](https://reviewable.io/reviews/cockroachdb/docs/11206#-MiYAKrkoeQo28MZLu-g-r1-6:-Micdlid6GIvXiwAN7pS:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/6088ab2fafe8a65fbed4ffd187e259216cbd1f50/_includes/v21.1/orchestration/start-cockroachdb-helm-secure.md#L6)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Not sure why this is a separate warning. Feels like it should be combined with the one above. 
</blockquote></details>

Done.

**Pull request merged**